yt/frontends/gadget/data_structures.py

```diff
@@ -189,6 +216,39 @@ def _calculate_offsets(self, field_list, pcounts):
             self._file_size,
         )
 
+    def _field_count(self, ptype, field):
+        # get the expected array size of the current particle type and field
+        count = self.total_particles[ptype]
+        if field in self.io._vector_fields:
+            count *= self.io._vector_fields[field]
+        return count
+
+    def _read_field(self, ptype, field, handle=None):
```

Comment:
We don't want to do this in this particular PR, but I want to at *some* point explore doing something slightly different here.

Right now we manually manage all of our memory.  What I would like to do is explore -- at initialization -- creating mmap handles to our file, and then accessing those mmap handles inside read field.